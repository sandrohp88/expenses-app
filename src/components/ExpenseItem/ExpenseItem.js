import React from 'react'
import moment from 'moment'
import { Link } from 'react-router-dom'
import numeral from 'numeral'
const ExpenseItem = ({ description, amount, createAt, id }) => (
  <div>
    <li>
      <Link to={`edit/${id}`}>
        <h3>{description}</h3>
      </Link>
      <p>
        {numeral(amount / 100).format('$0,0.00')}-
        {moment(createAt).format('MMMM Do YYYY')}
      </p>
    </li>
  </div>
)

export default ExpenseItem
