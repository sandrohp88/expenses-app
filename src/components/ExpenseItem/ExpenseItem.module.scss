@import '../../goblalStyles/content-container';

.listItem {
  display: flex;
  flex-direction: column;
  border: 1px solid darken($off-white, 7%);
  border-top: none;
  // border-radius: 6px;

  color: #272525;
  padding: $s-size;
  text-decoration: none;
  transition: background 0.3s ease;
  &:hover {
    background: $off-white;
  }

  @media (min-width: $desktop-breakpoint) {
    flex-direction: row;
    justify-content: space-between;
    padding: $m-size;
    align-items: center;
  }
}
.listItem__body {
  margin-bottom: $s-size;
  @media (min-width: $desktop-breakpoint) {
    margin-bottom: $m-size;
  }
}
.listItem__title {
  margin: 0;
  word-break: break-all;
}

.listItem__subtitle {
  color: $gray;
  font-size: 1.4rem;
  font-style: italic;
}
.listItem__data {
  margin: $s-size 0 0 0;
  @media (min-width: $desktop-breakpoint) {
    margin: 0;
    padding-left: $s-size;
  }
}
