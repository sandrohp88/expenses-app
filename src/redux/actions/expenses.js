import expensesRef from '../../firebase/firebase'
// ADD_EXPENSE
const addExpense = expense => ({
  type: 'ADD_EXPENSE',
  expense
})

// ADD_EXPENSE_ASYNC
const addExpenseAsync = (expenseData = {}) => {
  return async dispatch => {
    const {
      description = '',
      note = '',
      amount = 0,
      createdAt = 0
    } = expenseData
    const expense = { description, note, amount, createdAt }
    const response = await expensesRef.add(expense)
    return dispatch(addExpense({ id: response.id, ...expense }))
  }
}

// REMOVE_EXPENSE
const removeExpense = (id = '') => ({ type: 'REMOVE_EXPENSE', id })

// EDIT_EXPENSE
const editExpense = (id, updates) => ({ type: 'EDIT_EXPENSE', id, updates })

// SET_EXPENSES
const setExpenses = expenses => ({ type: 'SET_EXPENSES', expenses })

// GET_EXPENSES_ASYNC
const getExpensesAsync = () => {
  return async dispatch => {
    const response = await expensesRef.get()
    const expenses = []
    response.forEach(expense => {
      const {
        description = '',
        note = '',
        amount = 0,
        createdAt = 0
      } = expense.data()
      expenses.push({id:expense.id, description, note, amount, createdAt })
    })
    return dispatch(setExpenses(expenses))
  }
}

export {
  getExpensesAsync,
  addExpense,
  addExpenseAsync,
  removeExpense,
  editExpense,
  setExpenses
}
