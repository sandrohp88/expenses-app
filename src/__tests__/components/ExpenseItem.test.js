import React from 'react'
import ExpenseItem from '../../components/ExpenseItem/ExpenseItem'
import { shallow } from 'enzyme'
import expenses from '../../sampleData/expenses'

test('render ExpenseItem', () => {
  const wrapper = shallow(<ExpenseItem {...expenses[0]} />)
  expect(wrapper).toMatchSnapshot()
})
