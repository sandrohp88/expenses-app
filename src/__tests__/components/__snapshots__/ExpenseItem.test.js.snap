// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render ExpenseItem 1`] = `
<div>
  <li>
    <Link
      replace={false}
      to="edit/1"
    >
      <h3>
        Gum
      </h3>
    </Link>
    <p>
      195
       - 
      Dec 31st 69
    </p>
  </li>
</div>
`;
